The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10688640274/job/29628947576) [(plain)](https://github.com/rust-lang/rust/commit/65e78db8d7b41ec40fd9b789a96df12fd3d18180/checks/29628947576/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.895 Building wheels for collected packages: reuse
#13 2.896   Building wheel for reuse (pyproject.toml): started
#13 3.145   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.146   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 3.146   Stored in directory: /tmp/pip-ephem-wheel-cache-1fyiagzt/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.149 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.549 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.550 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.106 Collecting virtualenv
#13 4.106 Collecting virtualenv
#13 4.143   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 4.226      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 70.5 MB/s eta 0:00:00
#13 4.277 Collecting distlib<1,>=0.3.7
#13 4.280   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 4.287      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 86.9 MB/s eta 0:00:00
#13 4.334 Collecting filelock<4,>=3.12.2
#13 4.338   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 4.372 Collecting platformdirs<5,>=3.9.1
#13 4.375   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 4.455 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.643 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 4.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      239552 kB
DirectMap2M:     9197568 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/d6c8169c186ab16a3404cd0d0866674018e8a19e/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-d6c8169c186ab16a3404cd0d0866674018e8a19e-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
##[endgroup]
fmt check
Diff in /checkout/library/core/src/marker.rs:299:
 ///
 /// This isn't always desired. For example, shared references (`&T`) can be copied regardless of
 /// whether `T` is `Copy`. Likewise, a generic struct containing markers such as [`PhantomData`]
-/// could potentially be duplicated with a bit-wise copy. 
+/// could potentially be duplicated with a bit-wise copy.
 ///
 /// ## What's the difference between `Copy` and `Clone`?
 ///
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/prelude/mod.rs" "/checkout/library/core/src/primitive.rs" "/checkout/library/core/src/marker.rs" "/checkout/library/core/src/error.rs" "/checkout/compiler/rustc_lint/src/nonstandard_style/tests.rs" "/checkout/compiler/rustc_lint/src/unused.rs" "/checkout/compiler/rustc_lint/src/internal.rs" "/checkout/compiler/rustc_lint/src/drop_forget_useless.rs" "/checkout/compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs" "/checkout/compiler/rustc_lint/src/impl_trait_overcaptures.rs" "/checkout/compiler/rustc_lint/src/traits.rs" "/checkout/compiler/rustc_lint/src/types.rs" "/checkout/compiler/rustc_lint/src/tests.rs" "/checkout/compiler/rustc_lint/src/async_fn_in_trait.rs" "/checkout/compiler/rustc_lint/src/passes.rs" "/checkout/compiler/rustc_lint/src/precedence.rs" "/checkout/library/core/src/task/ready.rs" "/checkout/library/core/src/task/poll.rs" "/checkout/library/core/src/task/mod.rs" "/checkout/library/core/src/task/wake.rs" "/checkout/library/core/src/macros/mod.rs" "/checkout/library/core/src/arch.rs" "/checkout/library/core/src/pin.rs" "/checkout/library/core/src/alloc/global.rs" "/checkout/library/core/src/alloc/layout.rs" "/checkout/library/core/src/alloc/mod.rs" "/checkout/library/core/src/tuple.rs" "/checkout/library/core/src/panic.rs" "/checkout/library/core/src/time.rs" "/checkout/library/core/src/io/borrowed_buf.rs" "/checkout/library/core/src/io/mod.rs" "/checkout/library/core/src/borrow.rs" "/checkout/compiler/rustc_passes/src/upvars.rs" "/checkout/compiler/rustc_passes/src/naked_functions.rs" "/checkout/compiler/rustc_passes/src/liveness.rs" "/checkout/compiler/rustc_passes/src/lib.rs" "/checkout/compiler/rustc_passes/src/abi_test.rs" "/checkout/compiler/rustc_passes/src/layout_test.rs" "/checkout/compiler/rustc_passes/src/entry.rs" "/checkout/compiler/rustc_passes/src/reachable.rs" "/checkout/compiler/rustc_passes/src/hir_stats.rs" "/checkout/compiler/rustc_passes/src/check_const.rs" "/checkout/compiler/rustc_passes/src/errors.rs" "/checkout/compiler/rustc_passes/src/weak_lang_items.rs" "/checkout/compiler/rustc_passes/src/diagnostic_items.rs" "/checkout/compiler/rustc_passes/src/check_attr.rs" "/checkout/compiler/rustc_passes/src/stability.rs" "/checkout/compiler/rustc_passes/src/loops.rs" "/checkout/compiler/rustc_passes/src/lib_features.rs" "/checkout/library/core/src/char/methods.rs" "/checkout/library/core/src/char/convert.rs" "/checkout/library/core/src/char/decode.rs" "/checkout/library/core/src/char/mod.rs" "/checkout/library/core/src/option.rs" "/checkout/library/core/src/cell.rs" "/checkout/compiler/rustc_passes/src/liveness/rwu_table.rs" "/checkout/compiler/rustc_passes/src/hir_id_validator.rs" "/checkout/compiler/rustc_passes/src/dead.rs" "/checkout/compiler/rustc_passes/src/debugger_visualizer.rs" "/checkout/compiler/rustc_passes/src/lang_items.rs" "/checkout/compiler/rustc_driver/src/lib.rs" "/checkout/compiler/rustc_incremental/src/assert_dep_graph.rs" "/checkout/compiler/rustc_incremental/src/lib.rs" "/checkout/library/core/src/prelude/common.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Sep  3 18:40:37 UTC 2024
  network time: Tue, 03 Sep 2024 18:40:37 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/marker.rs
+++ b/library/core/src/marker.rs
@@ -299,7 +299,7 @@ marker_impls! {
 ///
 /// This isn't always desired. For example, shared references (`&T`) can be copied regardless of
 /// whether `T` is `Copy`. Likewise, a generic struct containing markers such as [`PhantomData`]
-/// could potentially be duplicated with a bit-wise copy. 
+/// could potentially be duplicated with a bit-wise copy.
 ///
 /// ## What's the difference between `Copy` and `Clone`?
 ///

--- a/library/core/src/marker.rs
+++ b/library/core/src/marker.rs
@@ -292,9 +292,10 @@ marker_impls! {
 /// bound on type parameters:
 ///
 /// ```
+/// #[derive(Clone)]
 /// struct MyStruct<T>;
 ///
-/// impl<T: Copy> Copy for MyStruct<T> { }
+/// impl<T: Copy> Copy for MyStruct<T> { } 
 /// ```
 ///
 /// This isn't always desired. For example, shared references (`&T`) can be copied regardless of

--- a/library/core/src/marker.rs
+++ b/library/core/src/marker.rs
@@ -293,7 +293,7 @@ marker_impls! {
 ///
 /// ```
 /// #[derive(Clone)]
-/// struct MyStruct<T>;
+/// struct MyStruct<T>(T);
 ///
 /// impl<T: Copy> Copy for MyStruct<T> { } 
 /// ```

